airbrakes/data_handling/data_processor.py

```diff
@@ -202,52 +202,111 @@ def _calculate_current_altitudes(self) -> npt.NDArray[np.float64]:
     def _calculate_rotated_accelerations(self) -> npt.NDArray[np.float64]:
         """
         Calculates the rotated acceleration vector. Converts gyroscope data into a delta
-        quaternion, and adds onto the last quaternion. Will most likely be replaced by IMU
-        quaternion data in the future, this is a work-around due to bad datasets.
+        quaternion, and adds onto the last quaternion. Only used when the quaternions are
+        deemed too uncertain to rely on.
 
-        :return: numpy list of rotated acceleration vector [x,y,z]
+        :return: numpy list of the vertical rotated accelerations
         """
-
         # We pre-allocate the space for our accelerations first
         rotated_accelerations = np.zeros(len(self._data_packets))
 
-        current_orientation = self._current_orientation_quaternions
-        # Iterates through the data points and time differences between the data points
-        for idx, (data_packet, dt) in enumerate(
-            zip(self._data_packets, self._time_differences, strict=True)
+        # checks for missing data
+        if any(
+            packet.estCompensatedAccelX is None
+            or packet.estCompensatedAccelY is None
+            or packet.estCompensatedAccelZ is None
+            for packet in self._data_packets
```

Comment:
```suggestion
```
fair to say if any of these are missing we are cooked. Better to just not include it it then.

Reply:
Yeah I was kinda debating this idea with what to check when running this function. It's not like we have ever encountered a situation where a packet doesn't have acceleration or gyroscope, so idk if it is even worth checking

Reply:
Yeah we have enough launches with this IMU to know that's not going to happen